cpp/src/cylon/arrow/arrow_comparator.cpp

```diff
@@ -16,14 +16,14 @@
 
 namespace cylon {
 
-template<typename ARROW_TYPE>
+template <typename ARROW_TYPE>
 class NumericArrowComparator : public ArrowComparator {
   int compare(const std::shared_ptr<arrow::Array> &array1, int64_t index1,
               const std::shared_ptr<arrow::Array> &array2, int64_t index2) override {
     auto reader1 = std::static_pointer_cast<arrow::NumericArray<ARROW_TYPE>>(array1);
     auto reader2 = std::static_pointer_cast<arrow::NumericArray<ARROW_TYPE>>(array2);
-
-    return (int) (reader1->Value(index1) - reader2->Value(index2));
+    auto diff = reader1->Value(index1) - reader2->Value(index2);
+    return (diff > 0) - (diff < 0);
```

Comment:
shall we add a comment here with a link to the bit twiddling hacks site?
https://graphics.stanford.edu/~seander/bithacks.html#CopyIntegerSign